perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2584,6 +2574,32 @@ private String getAttributeValueAsString (PerunSession session, User userToBeSpo
 		return attributeValue;
 	}
 
+	/**
+	 * Checks, if there is already a user with given login. If there is, throws the
+	 * InvalidSponsoredUserDataException.
+	 *
+	 * @param sess session
+	 * @param login login value
+	 * @param loginAttributeName login attribute urn
+	 * @throws InvalidSponsoredUserDataException if there is already a user with given login
+	 */
+	private void checkIfLoginIsFree(PerunSession sess, String login, String loginAttributeName) throws InvalidSponsoredUserDataException {
```

Comment:
I think we have similar logic somewhere (users manager?). If not, it should handle (check) also reserved logins from registrations, because they are not stored in attributes.

Reply:
Good point, for some reason, I overlooked it.